compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -4338,6 +4344,7 @@ void OMR::ValuePropagation::transformConverterCall(TR::TreeTop *callTree)
 
          case TR::sun_nio_cs_US_ASCII_Encoder_encodeASCII: threshold = 0; break;
          case TR::sun_nio_cs_US_ASCII_Decoder_decodeASCII: threshold = 0; break;
+         case TR::java_lang_StringCoding_implEncodeAsciiArray: threshold = 0; break;
```

Comment:
Something that already existed in the code. I assume original intention was to have different threshold for different methods, but looking at the values set, should we refactor this to have methods that needs threshold 0 combined ?

Reply:
How about the following reduction?
```
int32_t threshold = 0;
if (rm == TR::sun_nio_cs_ext_SBCS_Encoder_encodeSBCS)
   {
   threshold = 11;
   }
```
We can put the names of all converters (and threshold result) in a comment in order to preserve the prior analysis done on these methods.

Reply:
@hzongaro @r30shah The above refactoring and all the remaining formatting changes are available here: https://github.com/eclipse/omr/commit/08ce7c02b3e51cfc25b05e33a82bc0f4e6a4199e.

Once you both approve I'll squash the above commit.

Reply:
> Once you both approve I'll squash the above commit.

Please go ahead and squash.  Thanks!

Reply:
Commits are squashed and I've added an environment option to selectively disable this transformation (for easier defect management/problem isolation). I'll remove the option once the code is baked into the builds for a few months.